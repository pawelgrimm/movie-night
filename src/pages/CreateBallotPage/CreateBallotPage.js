import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import MovieSearch from "../../containers/MovieSearch/MovieSearch";
import MovieList from "./components/CreateBallotMovieList";
import ConfirmationModal from "../../components/ConfirmationModal/ConfirmationModal";
import { clearBallot, saveBallot } from "../../services/ballot/actions";
import WelcomeMessage from "../../components/WelcomeMessage/WelcomeMessage";
import Loader from "../../components/Loader/Loader";
import AppPage from "../../components/AppPage/AppPage";
import ButtonGroup from "../../components/ButtonGroup/ButtonGroup";
import { SaveBallotButton } from "./components/SaveBallotButton";

const mapStateToProps = ({ ballot }) => ({
  movies: ballot.movies,
  isSavingBallot: ballot.isSaving,
  newBallot: ballot.id,
});

const mapDispatchToProps = (dispatch) => ({
  clearBallot: () => dispatch(clearBallot()),
  saveBallot: () => dispatch(saveBallot()),
});

export const CreateBallotPage = ({
  appElement,
  clearBallot,
  movies = [],
  isSavingBallot,
  newBallot,
  saveBallot,
}) => {
  const [isConfirmationModalOpen, setConfirmationModalOpen] = useState(false);
  const history = useHistory();

  useEffect(() => {
    if (newBallot) {
      clearBallot();
      history.push(`ballot/${newBallot}`);
    }
  }, [newBallot]);

  return (
    <AppPage>
      <WelcomeMessage isHidden={movies.length > 0} />
      <MovieSearch numResults={5} />
      {movies.length > 0 && <MovieList />}
      <ButtonGroup>
        <SaveBallotButton />
        <button
          className="button button--secondary"
          onClick={() => setConfirmationModalOpen(true)}
          disabled={movies.length < 1}
        >
          Start Over
        </button>
      </ButtonGroup>
      <ConfirmationModal
        appElement={appElement}
        isOpen={isConfirmationModalOpen}
        onClose={() => setConfirmationModalOpen(false)}
        onConfirm={() => {
          clearBallot();
          setConfirmationModalOpen(false);
        }}
      />
    </AppPage>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateBallotPage);
