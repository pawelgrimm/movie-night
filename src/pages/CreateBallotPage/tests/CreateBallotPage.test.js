import { expect } from "@jest/globals";
import React from "react";
import { shallow } from "enzyme";
import { CreateBallotPage } from "../CreateBallotPage";
import { movies } from "../../../global/tests/fixtures/fixtures";

test("should render NominationPage with no movies", () => {
  const wrapper = shallow(
    <CreateBallotPage appElement={undefined} clearMovieList={() => {}} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render NominationPage with movies", () => {
  const wrapper = shallow(
    <CreateBallotPage
      appElement={undefined}
      clearMovieList={() => {}}
      movies={movies}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
