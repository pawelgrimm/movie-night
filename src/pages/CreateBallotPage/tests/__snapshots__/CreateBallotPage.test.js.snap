// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NominationPage with movies 1`] = `
<AppPage>
  <WelcomeMessage
    isHidden={true}
  />
  <MovieSearch
    numResults={5}
  />
  <CreateBallotMovieListContainer />
  <ButtonGroup>
    <SaveBallotButton />
    <StartOverButton />
  </ButtonGroup>
</AppPage>
`;

exports[`should render NominationPage with no movies 1`] = `
<AppPage>
  <WelcomeMessage
    isHidden={false}
  />
  <MovieSearch
    numResults={5}
  />
  <CreateBallotMovieListContainer />
  <ButtonGroup>
    <SaveBallotButton />
    <StartOverButton />
  </ButtonGroup>
</AppPage>
`;
